### DISCUSSION TITLE
Help. Preview build time normally 4-5m now taking 21m+ and counting...

### DISCUSSION THREAD

**Discussion by teddis** [vercel/community#6905] [Help]
### Summary

1d ago our builds took ~4 to 5m, consistently for months. AFAIK we have not made many or drastic changes since yesterday. Here's my build log:

Running build in Washington, D.C., USA (East) – iad1
--
15:00:05.905 | Cloning github.com/teddis/wildeye (Branch: dev, Commit: 88a4136)
15:00:07.151 | Cloning completed: 1.245s
15:00:22.305 | Restored build cache
15:00:22.388 | Running "vercel build"
15:00:23.735 | Vercel CLI 34.2.0
15:00:25.700 | > Detected Turbo. Adjusting default settings...
15:00:25.995 | Running "install" command: `yarn install`...
15:00:26.492 | yarn install v1.22.19
15:00:26.616 | [1/4] Resolving packages...
15:00:28.467 | warning Resolution field "@types/react@18.2.74" is incompatible with requested version "@types/react@18.2.66"
15:00:28.770 | [2/4] Fetching packages...
15:01:19.916 | warning chart.js@4.4.2: The engine "pnpm" appears to be invalid.
15:01:19.923 | [3/4] Linking dependencies...
15:01:19.925 | warning "@wildeye/cacheproxycurl > @commander-js/extra-typings@12.0.1" has incorrect peer dependency "commander@~12.0.0".
15:01:19.926 | warning " > @wildeye/frontend@2.2.0" has unmet peer dependency "react@^17.0.0 \|\| ^18.0.0".
15:01:19.927 | warning " > @wildeye/frontend@2.2.0" has unmet peer dependency "react-dom@^17.0.0 \|\| ^18.0.0".
15:01:45.648 | [4/4] Building fresh packages...
15:08:16.816 | Detected Next.js version: 14.2.3
15:08:16.817 | Running "turbo run build"
15:08:17.042 |  
15:08:17.042 | Attention:
15:08:17.042 | Turborepo now collects completely anonymous telemetry regarding usage.
15:08:17.042 | This information is used to shape the Turborepo roadmap and prioritize features.
15:08:17.042 | You can learn more, including how to opt-out if you'd not like to participate in this anonymous program, by visiting the following URL:
15:08:17.042 | https://turbo.build/repo/docs/telemetry
15:08:17.043 |  
15:08:17.229 | • Packages in scope: @wildeye/frontend
15:08:17.229 | • Running build in 1 packages
15:08:17.229 | • Remote caching enabled
15:08:17.603 | @wildeye/frontend:build: cache miss, executing 745a64dcaa82e4e8
15:08:18.240 | @wildeye/frontend:build: yarn run v1.22.19
15:08:18.412 | @wildeye/frontend:build: $ next build
15:09:54.688 | @wildeye/frontend:build:   ▲ Next.js 14.2.3
15:09:55.735 | @wildeye/frontend:build:   - Experiments (use with caution):
15:09:57.106 | @wildeye/frontend:build:     · instrumentationHook
15:09:57.316 | @wildeye/frontend:build:
15:10:15.448 | @wildeye/frontend:build:    Creating an optimized production build ...

It is 3:23 pm MST and have not seen a log entry for 13m now. How do I determine if this is a problem with my source or Vercel infrastructure? Should I try to restart the build?

I had to cancel this deployment after 30m30s.

Urgent...

Here's a link to my current deployment build: https://vercel.com/wildeye-team/wildeye/6rF5qdmhw6Zd9Yf7NK7Qj3uHJJab

### Steps to Reproduce

Just pushed my dev branch that triggers a normal Preview deployment.

**Comment by versecafe** [comment]
you seem to have a large jump here do you know what goes on between package builds and the app build?
```
15:01:45.648 | [4/4] Building fresh packages...
15:08:16.816 | Detected Next.js version: 14.2.3
```

**Comment by teddis** [comment]
@versecafe After much investigation, turned out to be a yarn install hang I had to chase down and this was the culprit.
